test/controllers/ImporterEORINumberControllerSpec.scala

```diff
@@ -0,0 +1,117 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.ControllerSpecBase
+import controllers.actions.FakeDataRetrievalAction
+import forms.ImporterEORINumberFormProvider
+import mocks.repositories.MockSessionRepository
+import models.UserAnswers
+import pages.ImporterEORINumberPage
+import play.api.http.Status
+import play.api.mvc.{AnyContentAsFormUrlEncoded, Result}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{charset, contentType, defaultAwaitTimeout, redirectLocation, status}
+import views.html.ImporterEORINumberView
+
+import scala.concurrent.Future
+
+class ImporterEORINumberControllerSpec extends ControllerSpecBase {
+
+  val userAnswersWithImporterEORINumber: Option[UserAnswers] = Some(UserAnswers("some-cred-id")
+    .set(
+      ImporterEORINumberPage, "GB345834921000"
+    ).success.value
+  )
+
+  private def fakeRequestGenerator(importerEORI: String): FakeRequest[AnyContentAsFormUrlEncoded] =
+    fakeRequest.withFormUrlEncodedBody(
+      "importerEORI" -> importerEORI
+    )
+
+  trait Test extends MockSessionRepository {
+    lazy val controller = new ImporterEORINumberController(
+      authenticatedAction,
+      dataRetrievalAction,
+      dataRequiredAction,
+      mockSessionRepository,
+      messagesControllerComponents,
+      form,
+      ImporterEORINumberView
+    )
+    private lazy val ImporterEORINumberView = app.injector.instanceOf[ImporterEORINumberView]
+    private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
+    val userAnswers: Option[UserAnswers] = Some(UserAnswers("some-cred-id"))
+    val formProvider: ImporterEORINumberFormProvider = injector.instanceOf[ImporterEORINumberFormProvider]
+    MockedSessionRepository.set(Future.successful(true))
+    val form: ImporterEORINumberFormProvider = formProvider
+  }
+
+  "GET onLoad" should {
+    "return OK" in new Test {
+      override val userAnswers: Option[UserAnswers] = Some(UserAnswers("credId"))
+      val result: Future[Result] = controller.onLoad()(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "return HTML" in new Test {
+      override val userAnswers: Option[UserAnswers] = Some(
+        UserAnswers("some-cred-id").set(ImporterEORINumberPage, "GB345834921000").success.value
+      )
+      val result: Future[Result] = controller.onLoad()(fakeRequest)
+      contentType(result) mustBe Some("text/html")
+      charset(result) mustBe Some("utf-8")
+    }
+  }
+
+  "POST onSubmit" when {
+
+    "payload contains valid data" should {
+
+      "return a SEE OTHER response when correct data with character values" in new Test {
+        override val userAnswers: Option[UserAnswers] = Some(UserAnswers("credId"))
```

Comment:
this overrides val userAnswers: Option[UserAnswers] = Some(UserAnswers("some-cred-id"))
which is the same empty user answers but with different id, what's the point here